library/src/main/java/com/crossbowffs/remotepreferences/RemotePreferenceProvider.java

```diff
@@ -143,13 +143,21 @@ public boolean onCreate() {
             if (file.isDeviceProtected() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
                 context = context.createDeviceProtectedStorageContext();
             }
-            SharedPreferences prefs = context.getSharedPreferences(file.getFileName(), Context.MODE_PRIVATE);
+            SharedPreferences prefs = getSharedPreferences(context, file.getFileName());
             prefs.registerOnSharedPreferenceChangeListener(this);
             mPreferences.put(file.getFileName(), prefs);
         }
         return true;
     }
 
+    /**
+     * Generate {@link SharedPreferences} to store the key-value data.
+     * Override this method to provide a custom implementation of {@link SharedPreferences}.
+     */
+    protected SharedPreferences getSharedPreferences(Context context, String name) {
```

Comment:
```suggestion
    protected SharedPreferences getSharedPreferences(Context context, String prefFileName) {
```

For consistency with the rest of the code and to make it a bit more obvious what 'name' refers to